pyanaconda/modules/common/structures/group.py

```diff
@@ -22,7 +22,7 @@
 
 __all__ = ["GROUP_GID_NOT_SET", "GroupData"]
 
-GROUP_GID_NOT_SET = -1
+GROUP_GID_NOT_SET = 4294967295
```

Comment:
This is a very unfriendly default. Also, we cannot be really sure that this value will be never valid in the future. Could you please, keep `-1` as a default and use `Int64`?

An alternative solution is to introduce new properties `gid_type` and `uid_type` that will return `DEFAULT` to indicate that the installer will use a default ID, or `VALUE` to indicate that the installer will use a specified ID.

Reply:
Yes, agreed. I just did s/UInt32/Int64/ to return this to the state before @martinpitt review. Basically it's now 2:1 that we want to waste four bytes so we can keep our -1.

Reply:
It's not really about waste -- that's trivial. It's about picking the data type that matches what's going on in the kernel, and that's 32 bit. It can neither be negative, nor be > 2³², so having an API that accepts either is just buggy.

Reply:
Also, `(uint32) -1` is exactly that magic number, so you can keep -1 even with uint32.

Reply:
I don't think the API really matters as much here. The whole user creation is just *wrong*. By design of the ancients, it must fail silently, and it must also keep backwards compatibility ad infinitum. Since there is no option to deprecate, it is impossible to change. *That* is the true reason why the value is not checked anywhere, it would be useless.

For that reason it's also pointless to have hard crashes when the untreated user inputs exceed the data type while sending the inputs over some transport layer (current state with `Int`). This change basically ensures the failures stay silent, same as the rest of the program does. The error surfaces with execution of `useradd` where it is caught and ignored.